search-app/src/cmr/search/services/humanizers/humanizer_report_service.clj

```diff
@@ -155,31 +147,41 @@
   []
   (stl-cache/create-single-thread-lookup-cache
    (fallback-cache/create-fallback-cache
-    (consistent-cache/create-consistent-cache
-     {:hash-timeout-seconds (report-cache-consistent-timeout-seconds)})
+    (consistent-cache/create-consistent-cache)
     (cubby-cache/create-cubby-cache))))
 
+(defn- create-and-save-humanizer-report
+  "Helper function to create the humanizer report, save it to the cache, and return the content."
+  [context]
+  (info "Generating humanizer report.")
+  (let [[report-generation-time humanizers-report] (util/time-execution
+                                                    (safe-generate-humanizers-report-csv context))]
+    (info (format "Humanizer report generated in %d ms." report-generation-time))
+    (cache/set-value (cache/context->cache context report-cache-key)
+                     csv-report-cache-key
+                     humanizers-report)
+    humanizers-report))
+
 (defn humanizers-report-csv
   "Returns a report of the humanizers currently used in collections as a CSV.
   If the report has not yet been generated by the job scheduler, a write is
   queued to the single-threaded cache followed immediately by a read for the
   newly cached data.
 
   This is the function that is called by the web service."
-  [context]
-  (cache/get-value (cache/context->cache context report-cache-key)
-                   csv-report-cache-key
-                   ;; If there is a cache miss, generate the report and then
-                   ;; return its value
-                   #(safe-generate-humanizers-report-csv context)))
+  [context regenerate?]
+  (if regenerate?
+    (create-and-save-humanizer-report context)
+    (cache/get-value (cache/context->cache context report-cache-key)
+                     csv-report-cache-key
+                     ;; If there is a cache miss, generate the report and then
+                     ;; return its value
+                     #(safe-generate-humanizers-report-csv context))))
 
 ;; A job for generating the humanizers report
 (defjob HumanizerReportGeneratorJob
-  [ctx system]
-  (info "Running scheduled job for generating the humanizer report ...")
-  (let [context {:system system}]
-    (cache/reset (cache/context->cache context report-cache-key))
```

Comment:
We had debated about that during the implementation. In fact, it had originally used `set-value`. 

👍